app/services/RestaurantService.java

```diff
@@ -172,61 +172,44 @@ public Restaurant getRestaurantWithId(final UUID id)
      * @return the popular restaurants
      */
     @SuppressWarnings("unchecked")
-    public List<Restaurant> getPopularRestaurants(UUID userId)
+    public PaginationAdapter<Restaurant> getPopularRestaurants(User user)
     {
+	if (user != null)
+	{
 
-	UserHabitsBean userHabits = getUserHabits(userId);
-      
-	List<Restaurant> restaurants = (List<Restaurant>) getSession().createCriteria(Restaurant.class)
-		.add(Restrictions.eq("city.id", userHabits.getCity()))
-		.add(Restrictions.eq("priceRange", userHabits.getUserPriceRange())).list();
+	    UserHabitsBean userHabits = getUserHabits(user);
+	    if (userHabits != null)
+	    {
 
-	List<Object> toRemove = new ArrayList<Object>();
+		List<Restaurant> restaurants = (List<Restaurant>) getSession().createCriteria(Restaurant.class)
+			.add(Restrictions.eq("city.id", userHabits.getCity()))
+			.add(Restrictions.eq("priceRange", userHabits.getUserPriceRange())).list();
+	
 
-	for (Restaurant restaurant : restaurants)
-	{
+		List<Restaurant> toRemove = new ArrayList<Restaurant>();
 
-	    if (restaurant.getAverageRating() < (userHabits.getUserRatingRange() - 0.5d)
-		    || restaurant.getAverageRating() > (userHabits.getUserRatingRange() + 0.5d))
-	    {
-		toRemove.add(restaurant);
-	    }
+		for (Restaurant restaurant : restaurants)
+		{
 
-	}
-	restaurants.removeAll(toRemove);
+		    if (restaurant.getAverageRating() < (userHabits.getUserRatingRange() - 0.5d)
+			    || restaurant.getAverageRating() > (userHabits.getUserRatingRange() + 0.5d))
+		    {
+			toRemove.add(restaurant);
+		    }
 
-	if (restaurants.size() > 0)
-	{
-	    return restaurants;
-	}
+		}
+		restaurants.removeAll(toRemove);
 
-	return new ArrayList<>();
-    }
-    @SuppressWarnings("unchecked")
-	public List<Restaurant> getPopularRestaurants() {
-
-		List<PopularRestaurantsBean> popularRestaurantsBeans = getSession().createCriteria(Reservation.class, "reservation")
-				.createAlias("reservation.table", "table")
-				.setProjection(Projections.projectionList()
-						.add(Projections.groupProperty("table.restaurantId").as("restaurantId"))
-						.add(Projections.count("table").as("tableCount")))
-				.addOrder(Order.asc("tableCount"))
-				.setResultTransformer(Transformers.aliasToBean(PopularRestaurantsBean.class))
-				.setMaxResults(6)
-				.list();
-
-		if (popularRestaurantsBeans.size() > 0) {
-			List<UUID> popularRestaurantsIds = popularRestaurantsBeans.stream().map(PopularRestaurantsBean::getRestaurantId).collect(Collectors.toList());
-
-			return (List<Restaurant>) getSession().createCriteria(Restaurant.class)
-					.add(Restrictions.in("id", popularRestaurantsIds))
-					.addOrder(Order.asc("name"))
-					.list();
+		if (restaurants.size() > 0)
+		{
+		    return PaginationAdapter.createOutput().setPageNumber(1)
+				.setPageSize(6).setModel(restaurants).setNumberOfPages(1l);
```

Comment:
This page size does not do anything because the list of restaurants can still be infinite.